email/src/main/java/edu/jhuapl/dorset/demos/EmailClient.java

```diff
@@ -0,0 +1,137 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.jhuapl.dorset.demos;
+
+import java.io.IOException;
+import java.util.Scanner;
+
+import javax.mail.Message;
+import javax.mail.MessagingException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+
+import edu.jhuapl.dorset.Application;
+import edu.jhuapl.dorset.Request;
+import edu.jhuapl.dorset.Response;
+import edu.jhuapl.dorset.agents.Agent;
+import edu.jhuapl.dorset.agents.DateTimeAgent;
+import edu.jhuapl.dorset.routing.Router;
+import edu.jhuapl.dorset.routing.SingleAgentRouter;
+
+public class EmailClient {
+
+    private static final Logger logger = LoggerFactory.getLogger(EmailClient.class);
+    private EmailManager manager;
+    private static Application app;
+
+    /**
+     * EmailClient
+     *
+     * @param manager   EmailManager object
+     */
+    public EmailClient(EmailManager manager) {
+        this.manager = manager;
+        Agent agent = new DateTimeAgent();
+        Router router = new SingleAgentRouter(agent);
+        app = new Application(router);
+    }
+
+    /**
+     * Main method in Email Client
+     * 
+     * @param args   command line arguments
+     * @throws MessagingException   if cannot connect to server
+     * @throws IOException   if errors occur while processing email
+     */
+    public static void main(String[] args) throws MessagingException, IOException {
+        Config config = ConfigFactory.load();
+        EmailClient client;
+        try {
+            client = new EmailClient(new EmailManager(config));
+        } catch (MessagingException e) {
+            throw new MessagingException("Could not connect to server. Check your connection and configurations. Quitting now.", e);
+        }
+        client.run();
+        client.close();
+    }
+
+    /**
+     * Process and respond to emails
+     *
+     * @throws MessagingException   if errors occur while processing email
+     * @throws IOException   if errors occur while processing email
+     */
+    private void run() throws MessagingException, IOException {
+        Scanner scan = new Scanner(System.in);
+
+        try {
+            System.out.println("Inbox: " + manager.getCount(FolderType.INBOX));
+            System.out.println("Type c to continue or q to quit: ");
+
+            while (!scan.nextLine().equals("q")) {
+                while (manager.getCount(FolderType.INBOX) > 0) {
+                    Message msg = manager.getMessage(FolderType.INBOX);
+                    String text = manager.readEmail(msg);
+                    if (text == null) { 
```

Comment:
I don't think you can get a null here anymore but an exception which will be handled by your try catch